consumer.go

```diff
@@ -251,25 +265,36 @@ func (c *Consumer) nackOnFailureToSend(delivery *rabbit.Delivery) {
 func (c *Consumer) sendToDelayQueue(retryCount int32, body []byte) error {
 	queueName := c.options.GetDelayQueueName()
 	c.logger.Infof("sending message to retry queue %s with retry count of %d", queueName, retryCount)
-	params := PublishParams{
-		Body:       body,
-		Headers:    map[string]interface{}{"retry_count": retryCount},
-		AmqpURL:    c.options.URL,
-		RoutingKey: queueName,
-	}
-	return PublishMessage(&params)
+
+	return c.channel.Publish(
+		"", // exchange
+		queueName,
+		false, // mandatory
+		false, // immediate
+		rabbit.Publishing{
+			ContentType: "text/plain",
+			Body:        body,
+			Headers: rabbit.Table(map[string]interface{}{
+				"retry_count": retryCount,
+			}),
+		})
 }
 
 func (c *Consumer) sendToDeadQueue(body []byte) error {
+	c.MessagesSentToDeadQueue++
```

Comment:
Is this tracked only for testing purposes?

Reply:
Yes.

Reply:
If that's so, I'd argue that maybe it would better to remove this. We can check how many messages are in the dead queue by consuming them, which might be a better test than checking a counter.